### PR TITLE
MGDOBR-1315: ShardAPI: GET Processors

### PR DESCRIPTION
**Pull request by manstis**
[https://issues.redhat.com/browse/MGDOBR-1315](https://issues.redhat.com/browse/MGDOBR-1315)

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [x] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [x] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-XXX] - $clear_explanation_of_what_you_did"
- [x] The layers in the `kustomize` folder have been updated accordingly.
- [x] All new functionality is tested
- [x] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

File: manager-parent/manager-v2/src/main/java/com/redhat/service/smartevents/manager/v2/services/ProcessorServiceImpl.java [hunk 1]
```diff
@@ -102,12 +103,6 @@ public Processor createProcessor(String bridgeId, String customerId, String owne
         return doCreateProcessor(bridge, customerId, owner, processorRequest);
     }
 
-    @Transactional
```
Comment (manstis) [inline manager-parent/manager-v2/src/main/java/com/redhat/service/smartevents/manager/v2/services/ProcessorServiceImpl.java:105]: I just moved this method as it was living in between two others that are related to each other.

File: manager-parent/manager-v2/src/main/java/com/redhat/service/smartevents/manager/v2/persistence/models/Processor.java [hunk 2]
```diff
@@ -31,9 +33,26 @@
...
+                        "left join (" +
+                        "  select processor_id, count(*) as conditions_count from Condition c where component='MANAGER' group by c.processor_id" +
+                        ") cp_counts on cp_counts.processor_id=p.id " +
+                        "left join (" +
+                        "  select processor_id, count(*)  as conditions_count from Condition c where component='MANAGER' and status='TRUE' group by c.processor_id" +
+                        "    ) cp_ready_counts on cp_ready_counts.processor_id=p.id " +
+                        "where " +
+                        "b.shard_id = :shardId and " +
+                        "cp_counts.conditions_count = cp_ready_counts.conditions_count")
+})
```
Comment (r00ta) [inline manager-parent/manager-v2/src/main/java/com/redhat/service/smartevents/manager/v2/persistence/models/Processor.java:55]: I think this query could be simplified with something like (just by heart, not tested at all)

```
@NamedNativeQueries({
        @NamedNativeQuery(name = "PROCESSOR_V2.findProcessorIdByShardIdToDeployOrDelete",
                query = "select p.id " +
                        "from Processor_V2 p " +
                        "left join Bridge_V2 b on p.bridge_id = b.id " +
                        "left join (" +
                        "  select processor_id, count(*) from Condition c where component='MANAGER' and status != 'TRUE' group by c.processor_id having count(*)=0" +
                        "    ) cp_ready on cp_ready.processor_id=p.id " +
                        "where " +
                        "b.shard_id = :shardId")
})
```

**Review by r00ta — COMMENTED** [review]
Looks great :thumbs

I have just one small idea on the query that might be simplified. Does that make sense?

File: manager-parent/manager-v2/src/main/java/com/redhat/service/smartevents/manager/v2/persistence/models/Processor.java (hunk 2 above)
Comment (manstis) [reply manager-parent/manager-v2/src/main/java/com/redhat/service/smartevents/manager/v2/persistence/models/Processor.java]: Thanks for the suggestion. Whilst it did not work when all `MANAGER` `Condition`'s are `TRUE` it pointed my towards the following which has one less `join` and one less sub-query so is an improvement on my original:
```
@NamedNativeQueries({
        @NamedNativeQuery(name = "PROCESSOR_V2.findProcessorIdByShardIdToDeployOrDelete",
                query = "select p.id " +
                        "from Processor_V2 p " +
                        "left join Bridge_V2 b on p.bridge_id = b.id " +
                        "left join (" +
                        "  select processor_id, count(*) as incomplete_count from Condition c where component='MANAGER' and status != 'TRUE' group by c.processor_id " +
                        "    ) cp on cp.processor_id = p.id " +
                        "where " +
                        "(cp.incomplete_count = 0 or cp.incomplete_count is null) and " +
                        "b.shard_id = :shardId")
```

**Review by r00ta — APPROVED** [review]
Looks good :+1: thanks!

**Review by winklerm — APPROVED** [review]
LGTM

**Pull request merged**